staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery_test.go

```diff
@@ -210,14 +215,400 @@ func TestOpenAPIDiskCache(t *testing.T) {
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedGroupNames:        []string{"", "apps"},
+			expectedGroupVersions:     []string{"v1", "apps/v1"},
+			expectedPreferredVersions: []string{"v1", "apps/v1"},
+		},
+		{
+			name: "Aggregated discovery: 1 group/1 version at /api, 1 group/2 versions at /apis",
+			corev1: &apidiscovery.APIGroupDiscoveryList{
+				Items: []apidiscovery.APIGroupDiscovery{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "",
+						},
+						Versions: []apidiscovery.APIVersionDiscovery{
+							{
+								Version: "v1",
+								Resources: []apidiscovery.APIResourceDiscovery{
+									discovery.CoreV1Pod,
+								},
+							},
+						},
+					},
+				},
+			},
+			apis: &apidiscovery.APIGroupDiscoveryList{
+				Items: []apidiscovery.APIGroupDiscovery{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "apps",
+						},
+						Versions: []apidiscovery.APIVersionDiscovery{
+							// v2 is preferred since it is first
+							{
+								Version: "v2",
+								Resources: []apidiscovery.APIResourceDiscovery{
+									discovery.AppsV2Deployment,
+								},
+							},
+							{
+								Version: "v1",
+								Resources: []apidiscovery.APIResourceDiscovery{
+									discovery.AppsV1Deployment,
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedGroupNames:        []string{"", "apps"},
+			expectedGroupVersions:     []string{"v1", "apps/v1", "apps/v2"},
+			expectedPreferredVersions: []string{"v1", "apps/v2"},
+		},
+		{
+			name:   "Aggregated discovery: /api returns nothing, 2 groups at /apis",
+			corev1: &apidiscovery.APIGroupDiscoveryList{},
+			apis: &apidiscovery.APIGroupDiscoveryList{
+				Items: []apidiscovery.APIGroupDiscovery{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "apps",
+						},
+						Versions: []apidiscovery.APIVersionDiscovery{
+							{
+								Version: "v1",
+								Resources: []apidiscovery.APIResourceDiscovery{
+									discovery.AppsV1Deployment,
+									discovery.AppsV1StatefulSet,
+								},
+							},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "batch",
+						},
+						Versions: []apidiscovery.APIVersionDiscovery{
+							// v1 is preferred since it is first
+							{
+								Version: "v1",
+								Resources: []apidiscovery.APIResourceDiscovery{
+									discovery.BatchV1Job,
+									discovery.BatchV1CronJob,
+								},
+							},
+							{
+								Version: "v1beta1",
+								Resources: []apidiscovery.APIResourceDiscovery{
+									discovery.BatchV1Beta1Job,
+									discovery.BatchV1Beta1CronJob,
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedGroupNames:        []string{"apps", "batch"},
+			expectedGroupVersions:     []string{"apps/v1", "batch/v1", "batch/v1beta1"},
+			expectedPreferredVersions: []string{"apps/v1", "batch/v1"},
+		},
+	}
+
+	for _, test := range tests {
+		// Create discovery cache dir
+		discoCache, err := os.MkdirTemp("", "")
```

Comment:
please use some kind of name for these so people know what they are if cleanup fails

Reply:
prior test also

Reply:
Is it absolutely necessary to use the disk to test this? It's not obvious to me, but it feels like this is testing more than is needed?